# Objective: Each subreddit must have 1000 unique selftexts
---

In [1]:
import pandas as pd

# Depression Data

In [2]:
depression = pd.read_csv('./datasets/depression.csv')

In [3]:
depression.head()

Unnamed: 0  approved_at_utc   subreddit  \
0           0              NaN  depression   
1           1              NaN  depression   
2           2              NaN  depression   
3           3              NaN  depression   
4           4              NaN  depression   

                                            selftext author_fullname  saved  \
0  We know that September 10 was World Suicide Pr...         t2_1t70  False   
1  Welcome to /r/depression's check-in post - a p...         t2_1t70  False   
2           I’m going to the movies. I’m so nervous…     t2_3y8evtoh  False   
3  Now i can save so i can get myself out of this...     t2_2mfcq6e9  False   
4  I always do this. I’ll stay up until the wee h...     t2_18u3sn83  False   

   mod_reason_title  gilded  clicked  \
0               NaN       0    False   
1               NaN       0    False   
2               NaN       1    False   
3               NaN       0    False   
4               NaN       0    False   

                                               title      ...        \
0  Reminder: NO ACTIVISM is allowed here at any t...      ...         
1                              Regular Check-In Post      ...         
2  I’m willingly leaving my house for the first t...      ...         
3                                i finally got a job      ...         
4  Does anyone else stay up super late to avoid b...      ...         

                                           permalink parent_whitelist_status  \
0  /r/depression/comments/9er34n/reminder_no_acti...                  no_ads   
1  /r/depression/comments/anh7tv/regular_checkin_...                  no_ads   
2  /r/depression/comments/c8ij3y/im_willingly_lea...                  no_ads   
3  /r/depression/comments/c8mnay/i_finally_got_a_...                  no_ads   
4  /r/depression/comments/c8m2cy/does_anyone_else...                  no_ads   

   stickied                                                url  \
0      True  https://www.reddit.com/r/depression/comments/9...   
1      True  https://www.reddit.com/r/depression/comments/a...   
2     False  https://www.reddit.com/r/depression/comments/c...   
3     False  https://www.reddit.com/r/depression/comments/c...   
4     False  https://www.reddit.com/r/depression/comments/c...   

   subreddit_subscribers   created_utc  discussion_type media  is_video  \
0                 503808  1.536613e+09              NaN   NaN     False   
1                 503808  1.549392e+09              NaN   NaN     False   
2                 503808  1.562121e+09              NaN   NaN     False   
3                 503808  1.562151e+09              NaN   NaN     False   
4                 503808  1.562147e+09              NaN   NaN     False   

  author_cakeday  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  

[5 rows x 97 columns]

## First Size of Depression Dataframe

In [4]:
depression.shape

(2501, 97)

## Length of Dataframe without duplicates

In [5]:
#i want to drop all selftext duplicates since this subreddit reposts blogs
unique = depression.drop_duplicates(subset = 'selftext')

In [6]:
#length of unique selftexts
len(unique)

923

In [7]:
unique.head()

Unnamed: 0  approved_at_utc   subreddit  \
0           0              NaN  depression   
1           1              NaN  depression   
2           2              NaN  depression   
3           3              NaN  depression   
4           4              NaN  depression   

                                            selftext author_fullname  saved  \
0  We know that September 10 was World Suicide Pr...         t2_1t70  False   
1  Welcome to /r/depression's check-in post - a p...         t2_1t70  False   
2           I’m going to the movies. I’m so nervous…     t2_3y8evtoh  False   
3  Now i can save so i can get myself out of this...     t2_2mfcq6e9  False   
4  I always do this. I’ll stay up until the wee h...     t2_18u3sn83  False   

   mod_reason_title  gilded  clicked  \
0               NaN       0    False   
1               NaN       0    False   
2               NaN       1    False   
3               NaN       0    False   
4               NaN       0    False   

                                               title      ...        \
0  Reminder: NO ACTIVISM is allowed here at any t...      ...         
1                              Regular Check-In Post      ...         
2  I’m willingly leaving my house for the first t...      ...         
3                                i finally got a job      ...         
4  Does anyone else stay up super late to avoid b...      ...         

                                           permalink parent_whitelist_status  \
0  /r/depression/comments/9er34n/reminder_no_acti...                  no_ads   
1  /r/depression/comments/anh7tv/regular_checkin_...                  no_ads   
2  /r/depression/comments/c8ij3y/im_willingly_lea...                  no_ads   
3  /r/depression/comments/c8mnay/i_finally_got_a_...                  no_ads   
4  /r/depression/comments/c8m2cy/does_anyone_else...                  no_ads   

   stickied                                                url  \
0      True  https://www.reddit.com/r/depression/comments/9...   
1      True  https://www.reddit.com/r/depression/comments/a...   
2     False  https://www.reddit.com/r/depression/comments/c...   
3     False  https://www.reddit.com/r/depression/comments/c...   
4     False  https://www.reddit.com/r/depression/comments/c...   

   subreddit_subscribers   created_utc  discussion_type media  is_video  \
0                 503808  1.536613e+09              NaN   NaN     False   
1                 503808  1.549392e+09              NaN   NaN     False   
2                 503808  1.562121e+09              NaN   NaN     False   
3                 503808  1.562151e+09              NaN   NaN     False   
4                 503808  1.562147e+09              NaN   NaN     False   

  author_cakeday  
0            NaN  
1            NaN  
2            NaN  
3            NaN  
4            NaN  

[5 rows x 97 columns]

### Since we have 923 unique self texts, lets go back to our Depression Data Collection notebook and get more blog posts in order to reach 1000 unique self texts 

## Trying Anxiety Subreddit


In [8]:
anxiety = pd.read_csv('./datasets/anxiety.csv')
anxiety.shape

(4999, 102)

In [9]:
#this is the length of unique selftexts from anxiety
unique = anxiety.drop_duplicates(subset = 'selftext')
len(unique)

970

#### We have a decent amount of unique blogs for the Anxiety Subreddit! Let's proceed with Anxiety!

# Conclusion

**The two subreddits we'll use for this project are: r/Anxiety and r/Depression**
- Let's export both dataframes with unique selftexts as csv files
- The four columns I want to keep are: subreddit, selftext, title, created_utc

### Anxiety 

In [10]:
anxiety.shape

(4999, 102)

In [11]:
unique_anxiety = anxiety.drop_duplicates(subset = 'selftext')

In [12]:
unique_anxiety.shape

(970, 102)

In [13]:
type(unique_anxiety)

pandas.core.frame.DataFrame

In [14]:
#creating unique anxiety df with selftext, title, and created utc
features = ['subreddit', 'selftext', 'title', 'created_utc']
unique_anxiety = unique_anxiety[features]

In [15]:
unique_anxiety.head()

subreddit                                           selftext  \
0   Anxiety  Greetings /r/Anxiety,\n\n\nI apologize for the...   
1   Anxiety  What have you accomplished this week? Share yo...   
2   Anxiety  I GOT AN OFFER!!!!!!\n\nI went back to pessimi...   
3   Anxiety  i have severe anxiety and it is really difficu...   
4   Anxiety  This may seem like a small victory or just com...   

                                               title   created_utc  
0                         CBD Megathread (June 2019)  1.560125e+09  
1                     Wednesday Wins - July 03, 2019  1.562156e+09  
2  [UPDATE] after almost canceling a job intervie...  1.562108e+09  
3  i went to a bookstore and got a coffee by myse...  1.562172e+09  
4  I just managed to shift from a heart rate of 1...  1.562172e+09

In [16]:
unique_anxiety.shape

(970, 4)

In [17]:
#creating unique anxiety df
unique_anxiety.to_csv('./datasets/unique_anxiety.csv')

### Depression

In [18]:
unique_depression = depression.drop_duplicates(subset = 'selftext')

In [19]:
unique_depression.shape

(923, 97)

In [20]:
#creating unique df for derpression 
features = ['subreddit', 'selftext', 'title', 'created_utc']
unique_depression = unique_depression[features]

In [21]:
unique_depression.head()

subreddit                                           selftext  \
0  depression  We know that September 10 was World Suicide Pr...   
1  depression  Welcome to /r/depression's check-in post - a p...   
2  depression           I’m going to the movies. I’m so nervous…   
3  depression  Now i can save so i can get myself out of this...   
4  depression  I always do this. I’ll stay up until the wee h...   

                                               title   created_utc  
0  Reminder: NO ACTIVISM is allowed here at any t...  1.536613e+09  
1                              Regular Check-In Post  1.549392e+09  
2  I’m willingly leaving my house for the first t...  1.562121e+09  
3                                i finally got a job  1.562151e+09  
4  Does anyone else stay up super late to avoid b...  1.562147e+09

In [22]:
unique_depression.shape

(923, 4)

In [23]:
#creating unique depression df
unique_depression.to_csv('./datasets/unique_depression.csv')